### ISSUE TITLE
Confusing error message when accessing text of content

### ISSUE THREAD

**Issue by TimotheAlbouy** [typst/typst#4307]
### Description

When you write the following code:

```
#let a = state("a", 3)
#let b = int((context a.get()).text)
```

you get the following error:

```
Error: content does not contain field "text"
   ╭─[/main.typ:9:32]
   │
 9 │ #let b = int((context a.get()).text)
───╯
```

I know that this is not good code, you would rather write `#let b = context a.get()`, but still, some users may write it and get an error saying that the content type does not have a text field, which is confusing because the content type DOES have a text field. For example, the following code works:

```
#let a = [3]
#let b = int(a.text)
#b
```

I would suggest changing this compiler error for something more precise.

### Use Case

Improving this error message of the compiler, which currently may confuse some people

**Comment by xkevio** [comment]
Well, the error is correct in this case though. `context` always returns some form of opaque content, hence why any usage of the needed values should be done inside the `context`-expression, and this opaque content does not have a `text` field.

Your second example is simple enough to have a `.text` field. It really depends as `content` can be a lot of things.

**Comment by laurmaedje** [comment]
Maybe we can try to come up with a general way to add a context hint on any error related to access on contextual content.

**Comment by PgBiel** [comment]
An error which suggests that the `context` scope needs to be expanded seems ideal, or at least potentially more helpful.

**Comment by astrale-sharp** [comment]
For the user, "context scope needs to be expanded" will not be helpful if it doesn't come with a hint for how to do that in my opinion.

What about, if content.fields is empty, we hint that the content is not inspectable, this might be because it was generated in a context scope which is not accessible anymore?

(Most of the time fields is not empty otherwise, no?)

**Comment by bluebear94** [comment]
There are a couple of elements other than context without any visible fields, so better to match on the type of the element.

**Comment by PgBiel** [comment]
> For the user, "context scope needs to be expanded" will not be helpful if it doesn't come with a hint for how to do that in my opinion.

I agree that particular string would be a pretty awful message. Maybe we can use something like "can't access values depending on context outside that context" or "values which depend on context must be accessed inside it" instead, which at least suggests one potential action.

> What about, if content.fields is empty, we hint that the content is not inspectable, this might be because it was generated in a context scope which is not accessible anymore?

I feel like this wouldn't generalize too well, e.g. for `counter.update` or `state.update`.
